centreon/src/Core/Dashboard/Application/UseCase/FindFavoriteDashboards/FindFavoriteDashboards.php

```diff
@@ -0,0 +1,304 @@
+use Core\Dashboard\Application\UseCase\FindFavoriteDashboards\Response\UserResponseDto;
+use Core\Dashboard\Domain\Model\Dashboard;
+use Core\Dashboard\Domain\Model\DashboardRights;
+use Core\Dashboard\Domain\Model\Role\DashboardSharingRole;
+use Core\Dashboard\Domain\Model\Share\DashboardContactGroupShare;
+use Core\Dashboard\Domain\Model\Share\DashboardContactShare;
+use Core\Dashboard\Domain\Model\Share\DashboardSharingRoles;
+use Core\Media\Domain\Model\Media;
+use Core\Security\AccessGroup\Application\Repository\ReadAccessGroupRepositoryInterface;
+use Core\Security\AccessGroup\Domain\Model\AccessGroup;
+use Core\UserProfile\Application\Repository\ReadUserProfileRepositoryInterface;
+
+final class FindFavoriteDashboards
+{
+    use LoggerTrait;
+    public const AUTHORIZED_ACL_GROUPS = ['customer_admin_acl'];
+
+    /** @var int[] */
+    private array $usersFavoriteDashboards;
+
+    /**
+     * @param RequestParametersInterface $requestParameters
+     * @param ReadDashboardRepositoryInterface $dashboardReader
+     * @param ReadUserProfileRepositoryInterface $userProfileReader
+     * @param ContactInterface $contact
+     * @param DashboardRights $rights
+     * @param ReadAccessGroupRepositoryInterface $readAccessGroupRepository
+     * @param ReadContactRepositoryInterface $readContactRepository
+     * @param ReadDashboardShareRepositoryInterface $readDashboardShareRepository
+     * @param bool $isCloudPlatform
+     */
+    public function __construct(
+        private readonly RequestParametersInterface $requestParameters,
+        private readonly ReadDashboardRepositoryInterface $dashboardReader,
+        private readonly ReadUserProfileRepositoryInterface $userProfileReader,
+        private readonly ContactInterface $contact,
+        private readonly DashboardRights $rights,
+        private readonly ReadAccessGroupRepositoryInterface $readAccessGroupRepository,
+        private readonly ReadContactRepositoryInterface $readContactRepository,
+        private readonly ReadDashboardShareRepositoryInterface $readDashboardShareRepository,
+        private readonly bool $isCloudPlatform,
+    ) {
+    }
+
+    /**
+     * @return FindFavoriteDashboardsResponse|ResponseStatusInterface
+     */
+    public function __invoke(): FindFavoriteDashboardsResponse|ResponseStatusInterface
+    {
+        try {
+            $profile = $this->userProfileReader->findByContact($this->contact);
+            $this->usersFavoriteDashboards = $profile !== null ? $profile->getFavoriteDashboards(): [];
+
+            if ($this->usersFavoriteDashboards === []) {
+                return new FindFavoriteDashboardsResponse([]);
+            }
+
+            // Hack to benifit from existing code.
+            $search = $this->requestParameters->getSearch();
+            $search = ['id' => ['$in' => $this->usersFavoriteDashboards], ...$search];
+            $this->requestParameters->setSearch(json_encode($search) ?: '');
```

Comment:
You must to test the usage of this method json_encode because it can return false. You can use the [JSON_THROW_ON_ERROR](https://www.php.net/manual/en/json.constants.php#constant.json-throw-on-error) flag to throw an exception instead for example (don't forget to catch the JsonException).

Reply:
done ! to valid with this PR : #6241 @adr-mo